Mathlib/Analysis/BoxIntegral/Basic.lean

```diff
@@ -669,13 +669,186 @@ end Integrable
 
 open MeasureTheory
 
+/-!
+### Oscillation
+-/
+
+section oscillation
+
+open EMetric
+
+/-- The oscillation of `f` at `x`. -/
```

Comment:
You should probably move all the results about `oscillation` to a new file, say`Analysis.Oscillation`. Then, you can generalize the results to `f : E \to F` with `E` a `TopologicalSpace E` and `F` a `PseudoEMetricSpace` or `MetricSpace` from my experience.